src/plan/immix/global.rs

```diff
@@ -81,21 +82,25 @@ impl<VM: VMBinding> Plan for Immix<VM> {
     fn schedule_collection(&'static self, scheduler: &GCWorkScheduler<VM>) {
         self.base().set_collection_kind::<Self>(self);
         self.base().set_gc_status(GcStatus::GcPrepare);
-        let in_defrag = self.immix_space.decide_whether_to_defrag(
-            self.is_emergency_collection(),
-            true,
-            self.base().cur_collection_attempts.load(Ordering::SeqCst),
-            self.base().is_user_triggered_collection(),
-            *self.base().options.full_heap_system_gc,
-        );
-
-        // The blocks are not identical, clippy is wrong. Probably it does not recognize the constant type parameter.
-        #[allow(clippy::if_same_then_else)]
-        if in_defrag {
-            scheduler.schedule_common_work::<ImmixGCWorkContext<VM, TRACE_KIND_DEFRAG>>(self);
-        } else {
-            scheduler.schedule_common_work::<ImmixGCWorkContext<VM, TRACE_KIND_FAST>>(self);
-        }
+        // let in_defrag = self.immix_space.decide_whether_to_defrag(
```

Comment:
We probably want to remove these commented out sections?

Reply:
Yes, I will. This is just a draft. If you are curious, you are welcome to look at the changes and leave any comments. But do not spend much time on a detailed review yet.